### PR TITLE
Add Formula for ahoy: A tool for making cli apps from yaml configs.

### PR DESCRIPTION
**Pull request by frankcarey**
See https://github.com/ahoy-cli/ahoy

Sponsored by Vimeo

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/ahoy.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  def install
+    ENV["GOPATH"] = buildpath
+    bin_path = buildpath/"src/github.com/ahoy-cli/ahoy"
+    bin_path.install Dir["*"]
+    cd bin_path do
+      system "go", "build", "-o", bin/"ahoy", "-ldflags", "-X main.version=#{version}", "."
+    end
+  end
+
+  def caveats; <<~EOPLIST
```
Comment (claui) [inline Formula/ahoy.rb]: Use `EOS` (end of string) instead of `EOPLIST`.

File: Formula/ahoy.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    See other changes at:
+
+    https://github.com/ahoy-cli/ahoy
+
+  EOPLIST
+  end
+
+  test do
+    system "#{bin}/ahoy", "--version"
```
Comment (claui) [inline Formula/ahoy.rb]: The test needs to exercise at least some of the program’s core features.

For example, you could write a minimal YAML with a single `command` named `hello`, run `ahoy` against it, and compare its standard output to `Hello world`. How about that?

File: Formula/ahoy.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class Ahoy < Formula
+  desc "Creates self documenting cli programs from commands in yaml files"
```
Comment (claui) [inline Formula/ahoy.rb]:
```suggestion
  desc "Creates self documenting CLI programs from commands in YAML files"
```

**Review by claui — CHANGES_REQUESTED** [review]

File: Formula/ahoy.rb (hunk 1 above)
Comment (frankcarey) [reply Formula/ahoy.rb]: done

File: Formula/ahoy.rb (hunk 2 above)
Comment (frankcarey) [reply Formula/ahoy.rb]: done

File: Formula/ahoy.rb (hunk 3 above)
Comment (frankcarey) [reply Formula/ahoy.rb]: done

**Comment by frankcarey** [comment]
@claui Thanks for the review!

**Review by claui — APPROVED** [review]

**Comment by claui** [comment]
Thanks @frankcarey for your contribution and for submitting a high-quality PR right from the start. 🍻

**Pull request closed without merging**